### PR TITLE
Remove `qualify_min_const_fn` pass

### PR DESCRIPTION
**Pull request by ecstatic-morse**
Having two MIR const-checking passes is confusing. This PR unifies `qualify_min_const_fn` and `check_consts`.

All checks from `qualify_min_const_fn` that were not already implemented in `check_consts` have been assigned a structured error. Unfortunately, this will require a lot of changes to expected test output. This is because `qualify_min_const_fn` bailed out after the first error was found, while `check_consts` reports *all* errors in the function.

cc @rust-lang/wg-const-eval

### CODE REVIEW

**Comment by Dylan-DPC** [comment]
@ecstatic-morse any updates?

**Comment by oli-obk** [comment]
r? @oli-obk

totally forgot this exists

**Comment by oli-obk** [comment]
let's wait for `const_if_match` first, this will get rid of a bunch of "unrelated" changes in this PR.

File: src/librustc_mir/transform/check_consts/ops.rs [hunk 1]
```diff
@@ -15,9 +17,9 @@ pub trait NonConstOp: std::fmt::Debug {
     /// Whether this operation can be evaluated by miri.
     const IS_SUPPORTED_IN_MIRI: bool = true;
 
-    /// Returns a boolean indicating whether the feature gate that would allow this operation is
-    /// enabled, or `None` if such a feature gate does not exist.
-    fn feature_gate(_tcx: TyCtxt<'tcx>) -> Option<bool> {
+    /// Returns the `Symbol` for the feature gate that would allow this operation, or `None` if
+    /// such a feature gate does not exist.
+    fn feature_gate() -> Option<Symbol> {
```
Comment (oli-obk) [inline src/librustc_mir/transform/check_consts/ops.rs:22]: This change would be better as its own independent PR, as then this PR really only does what it advertises in the title.

Comment (ecstatic-morse) [reply src/librustc_mir/transform/check_consts/ops.rs:22]: See #70013. This PR is quite old. The main blocker is blessing a bunch of `//~ ERROR` annotations.

**Comment by ecstatic-morse** [comment]
Closing this, but I think I'll get to it in the near future.
